### PR TITLE
WIP Output: Replace log_printers with built-in logging

### PR DESCRIPTION
**Pull request by aptrishu**
It replaces log_printers from ConsoleInteraction.py,
Processing.py and some other places with built-in
logging.

Closes https://github.com/coala/coala/issues/3472

### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

**After you submit your pull request, DO NOT click the 'Update Branch' button**

### CODE REVIEW

**Comment by aptrishu** [comment]
@sils are you expecting something like this ?

File: coalib/output/ConsoleInteraction.py [hunk 1]
```diff
@@ -373,26 +368,24 @@ def print_affected_files(console_printer,
             if (
                     sourcerange.file is not None and
                     sourcerange.file not in file_dict):
-                log_printer.warn('The context for the result ({}) cannot '
-                                 'be printed because it refers to a file '
-                                 "that doesn't seem to exist ({})"
-                                 '.'.format(result, sourcerange.file))
+                warnings.warn('The context for the result ({}) cannot '
```
Comment (nemaniarjun) [inline coalib/output/ConsoleInteraction.py]: I dont think we need this. `Logging.warn` should work fine :D

Comment (aptrishu) [inline coalib/output/ConsoleInteraction.py]: No, actually we're not checking the logs for this specifically (also no tests are there for that) so I have used warnings here (It doesn't log the message). As you see its kind-of demo PR, so I'll modify the tests sometime later. :P

**Comment by supergr35** [comment]
ping @aptrishu

**Comment by aptrishu** [comment]
Pong @supergr35 
This PR will take sometime, deprecation things are messy.

**Comment by aptrishu** [comment]
@sils could you please review it, if I'm going in the right direction.

**Comment by shreyans800755** [comment]
@aptrishu Are you still working on this issue ?
If not, can you please unassign yourself through gitter ?
There doesn't seem to be any activity on it for almost 3-4 months ?
Or if any maintainer can unassign so that others can work on the issue provided no work in going on for this.
CC @sils

**Comment by aptrishu** [comment]
cool @shreyans800755 , you can take it up if you want. I have unassigned myself.

File: coalib/output/ConsoleInteraction.py [hunk 2]
```diff
@@ -347,20 +344,18 @@ def print_results_formatted(log_printer,
                                         severity_str=severity_str,
                                         **format_args))
         except KeyError as exception:
-            log_printer.log_exception(
-                'Unable to print the result with the given format string.',
-                exception)
+            logging.exception(
+                'Unable to print the result with the given format string.{}'
+                .format(exception))
```
Comment (thisch) [inline coalib/output/ConsoleInteraction.py]: logging.exception implicitly adds the traceback to the log message. See the output of 

```
import logging
try:
    1/0
except Exception as e:
    logging.exception("message")
```

**Comment by thisch** [comment]
@shreyans800755 are you still interested in this? If not, I'll take this over

**Comment by shreyans800755** [comment]
@thisch I'm working on it.
I've made considerable changes in code and trying to pass tests. I'll push my code and create PR soon.
Its fairly big task, as log_printer is used quite extensively. So, it might take up some time.
Meanwhile, its better to close this PR as there shouldn't be any work here as confirmed here https://github.com/coala/coala/pull/3633#issuecomment-302946023.

**Comment by thisch** [comment]
Sry @sahilarora535, I didn't see that you are assigned to work on #3472 by the cobot.

**Comment by Makman2** [comment]
Taken over by @shreyans800755 :+1:

**Pull request closed without merging**